File: mobile/src/main/java/com/windscribe/mobile/networksecurity/NetworkSecurityPresenter.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.networksecurity;

import android.content.Context;

import com.windscribe.vpn.localdatabase.tables.NetworkInfo;

public interface NetworkSecurityPresenter {

    String getSavedLocale();

    void onAdapterSet();

    void onDestroy();

    void onNetworkSecuritySelected(NetworkInfo networkInfo);

    void onCurrentNetworkClick();

    void setTheme(Context context);

    void setupNetworkListAdapter();

    void init();

    void onAutoSecureToggleClick();
}

File: mobile/src/main/java/com/windscribe/mobile/networksecurity/NetworkSecurityPresenter.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.mobile.networksecurity

import android.content.Context
import com.windscribe.vpn.localdatabase.tables.NetworkInfo

interface NetworkSecurityPresenter {
    val savedLocale: String
    fun onAdapterSet()
    fun onDestroy()
    fun onNetworkSecuritySelected(networkInfo: NetworkInfo)
    fun onCurrentNetworkClick()
    fun setTheme(context: Context)
    fun setupNetworkListAdapter()
    fun init()
    fun onAutoSecureToggleClick()
}